operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/AbstractDependentResource.java

```diff
@@ -71,9 +71,12 @@ protected ReconcileResult<R> reconcile(P primary, R actualResource, Context<P> c
             logForOperation("Updating", primary, desired);
             var updatedResource = handleUpdate(actualResource, desired, primary, context);
             return ReconcileResult.resourceUpdated(updatedResource);
+          } else {
+            log.debug("Update skipped for dependent {} as it matched the existing one",
+                actualResource);
```

Comment:
Let's not log the whole resources, this is a security concern, to not log secrets, thus sensitive data.